proof/access-control/Syscall_AC.thy

```diff
@@ -438,95 +396,223 @@ lemma ethread_set_time_slice_pas_refined[wp]:
   done
 
 lemma thread_set_time_slice_pas_refined[wp]:
-  "\<lbrace>pas_refined aag\<rbrace>
-     thread_set_time_slice tptr time
-   \<lbrace>\<lambda>rv. pas_refined aag\<rbrace>"
-  apply (simp add: thread_set_time_slice_def | wp)+
-  done
+  "thread_set_time_slice tptr time \<lbrace>pas_refined aag\<rbrace>"
+  by (simp add: thread_set_time_slice_def | wp)+
 
 lemma dec_domain_time_pas_refined[wp]:
-  "\<lbrace>pas_refined aag\<rbrace>
-     dec_domain_time
-   \<lbrace>\<lambda>rv. pas_refined aag\<rbrace>"
+  "dec_domain_time \<lbrace>pas_refined aag\<rbrace>"
   apply (simp add: dec_domain_time_def | wp)+
   apply (clarsimp simp: pas_refined_def tcb_domain_map_wellformed_aux_def)
   done
 
 crunch pas_refined[wp]: timer_tick "pas_refined aag"
   (wp_del: timer_tick_extended.pas_refined_tcb_domain_map_wellformed)
 
+
+locale Syscall_AC_wps =
+  fixes f :: "(det_ext state, 'z) nondet_monad"
+  and aag :: "'a PAS"
+  assumes respects[wp]: "f \<lbrace>integrity aag X st\<rbrace>"
+  and pas_refined[wp]: "f \<lbrace>pas_refined aag\<rbrace>"
+
+locale Syscall_AC_1 =
+  fixes aag :: "'a PAS"
+  assumes invs_irq_state_update[simp]:
+    "invs ((s :: det_ext state)\<lparr>machine_state := irq_state_update f s'\<rparr>) =
+     invs (s\<lparr>machine_state := s'\<rparr>)"
+  and prepare_thread_delete_Syscall_AC_wps''[simp]:
```

Comment:
At some point (not in this PR), we should think about a mechanism that lets us express that we want a predicate to hold for a whole bunch of functions. Might be as simple as combinator that takes a list and lets you automatically derive the single predicates from that list.